bond/util/validation.go

```diff
@@ -0,0 +1,70 @@
+package util
+
+import (
+	"crypto/rand"
+	"fmt"
+	"github.com/vishvananda/netlink"
+)
+
+func HandleMacDuplicates(linkObjectsToBond []netlink.Link, netNsHandle *netlink.Handle) error {
+	macsInUse := []string{}
+	var err error
+	for _, link := range linkObjectsToBond {
+		linkMac := link.Attrs().HardwareAddr.String()
+		if isMacDuplicated(linkMac, macsInUse) {
+			linkMac, err = updateDuplicateMac(link, netNsHandle, macsInUse)
+			if err != nil {
+				return err
+			}
+		}
+		macsInUse = append(macsInUse, linkMac)
+	}
+	return nil
+}
+
+func isMacDuplicated(mac string, macsInUse []string) bool {
+	for _, usedMac := range macsInUse {
+		if mac == usedMac {
+			return true
+		}
+	}
+	return false
+}
+
+func updateDuplicateMac(link netlink.Link, netNsHandle *netlink.Handle, macsInUse []string) (string, error) {
+	var newMac string
+	var err error
+	// Setting of mac sometimes fails for various reasons. Retry a few times before failing.
```

Comment:
This doesn't seem like a good way of getting around a failure.

What if it fails more than 10 times?

What are the various reasons and can we fix them?

Reply:
Seems like it fails for specific macs. I tried to repeat this with just one mac, but then it fails repeatedly. Choosing a different mac seems to help.

Reply:
What are the specific macs, could we filter them out when generating the random macs?
Or at the very least, check if the mac fails and repeat indefinitely with a timeout, add better logging and create an issue to look at this in the future.

Reply:
Limiting this to local/unicast seems to help.

Reply:
LGTM